### ISSUE TITLE
Un-Pythonic escapes and escape sequences in PEG regexs and string literals

### ISSUE THREAD

**Issue by smbolton** [igordejanovic/Arpeggio#32]
Hi Igor! Thanks for arpeggio, I'm really enjoying using it.

This could either be a bug report, or a feature/documentation request, since "fixing" the issue may break some existing PEG grammars.

When I started writing PEG grammars for arpeggio, I assumed they would use regular Python syntax regarding escapes and escape sequences in regexs and string literals. Then I discovered several things that don't work that way (they can be done using Python grammars, of course):

This fails by matching too much, if any single quotes occur after it in the grammar:
~~~
  rule = '\\'  # literal backslash before final single quote
~~~
As a work-around, one can use "\\", since backslash escape isn't implemented for double-quoted strings. But that also means this doesn't work:
~~~
  rule = "a\"b"  # escaped double quote inside double-quoted string
~~~
Regular expression recognition is similarly limited:
~~~
  rule = r'\\[abc]\\'  # backslash before final single quote can fail
  
  rule = r"."   # double quotes not supported at all
~~~
Once string literals are parsed, escape sequences within them are incorrectly converted:
~~~
  rule = '\\n\n'  # matches 'backslash-newline-newline', not 'backslash-n-newline'.

  rule = '\a\b\f\r\v\101\x41\u266f\U0001036a'   # none are converted as one might expect
~~~
I can send you pull requests that will change quote-escaping to match the usual Python rules, and implement Python-standard escape sequence recognition. But first, I wanted to get your thoughts on the issue. While the changes I'd like to see follow the principle of least surprise, and would enhance the capability of PEG grammars, they probably will break at least a few grammars existing in the wild.

Thanks again,

-Sean

**Comment by igordejanovic** [comment]
Hi Sean,
This looks like a bug to me. I'm not aware of any PEG standard for handling escape sequences so I agree that it would be best to follow Python rules. Go ahead and send me PR. I believe that this affects both cleanpeg and peg modules. If you could put these cases in regression tests that would be great. See `/tests/unit/regressions/`.
Thanks.

**Comment by smbolton** [comment]
Sigh. Now why didn't my pull request get attached to this issue?

**Comment by igordejanovic** [comment]
Fixed in PR #33.
Thanks Sean!
